app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx

```diff
@@ -0,0 +1,277 @@
+import classNames from 'classnames/bind';
+import * as COLORS from 'common/constants/colors';
+import ReactDOMServer from 'react-dom/server';
+import { TooltipWrapper } from '../common/tooltip';
+import { C3Chart } from '../common/c3chart';
+import styles from './launchExecutionAndIssueStatistics.scss';
+import { Legend } from './launchExecutionAndIssuesChartLegend';
+import { LaunchExecutionAndIssueStatisticsTooltip } from './launchExecutionAndIssueStatisticsTooltip';
+import { getPercentage, getDefectItems } from './chartUtils';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  statistics$executions$total: {
+    id: 'FilterNameById.statistics$executions$total',
+    defaultMessage: 'Total',
+  },
+  statistics$executions$passed: {
+    id: 'FilterNameById.statistics$executions$passed',
+    defaultMessage: 'Passed',
+  },
+  statistics$executions$failed: {
+    id: 'FilterNameById.statistics$executions$failed',
+    defaultMessage: 'Failed',
+  },
+  statistics$executions$skipped: {
+    id: 'FilterNameById.statistics$executions$skipped',
+    defaultMessage: 'Skipped',
+  },
+});
+
+@injectIntl
+export class LaunchExecutionChart extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool,
+    container: PropTypes.instanceOf(Element).isRequired,
+    observer: PropTypes.object.isRequired,
+  };
+
+  static defaultProps = {
+    isPreview: false,
+    height: 0,
+  };
+
+  state = {
+    isConfigReady: false,
+  };
+
+  componentDidMount() {
+    this.props.observer.subscribe('widgetResized', this.resizeStatusChart);
+    this.getConfig();
+  }
+  componentWillUnmount() {
+    this.statusNode.removeEventListener('mousemove', this.getCoords);
+    this.props.observer.unsubscribe('widgetResized', this.resizeStatusChart);
+  }
+
+  onStatusChartCreated = (chart, element) => {
+    this.statusChart = chart;
+    this.statusNode = element;
+    if (!this.props.widget.content.result || this.props.isPreview) {
+      return;
+    }
+
+    d3
+      .select(chart.element)
+      .select('.c3-chart-arcs-title')
+      .attr('dy', -15)
+      .append('tspan')
+      .attr('dy', 30)
+      .attr('x', 0)
+      .attr('fill', '#666')
+      .text('SUM');
+
+    this.statusNode.addEventListener('mousemove', this.getCoords);
+  };
+
+  onStatusMouseOut = () => {
+    this.statusChart.revert();
+  };
+
+  onStatusMouseOver = (id) => {
+    this.statusChart.focus(id);
+  };
+
+  onStatusClick = (id) => {
+    this.statusChart.toggle(id);
+  };
+
+  getConfig = () => {
+    const { widget, container } = this.props;
+
+    this.height = container.offsetHeight;
+    this.width = container.offsetWidth;
+    this.noAvailableData = false;
+    const statusChartData = {};
+    const statusChartDataOrdered = [];
+    const statusChartColors = {};
+    const values = widget.content.result[0].values;
+    Object.keys(values).forEach((key) => {
```

Comment:
Can we reuse this logic? Now it's totally copy-paste.

Reply:
we could but I think the effort for making a 3 lines long code that is highly similar into some generic function with a bunch of added parameters is not justified

Reply:
Ok, let's leave it as is.